### PR TITLE
OQS-213 add pre-commit setup

### PR DESCRIPTION
**Pull request by renefritze**

### CODE REVIEW

**Comment by renefritze** [comment]
This PR needs to be admin-merged since the JIRA check is broken (at least for this PR) and will only be fixed in https://github.com/arup-group/actions-composite-oasys-jira-check/pull/6

File: .github/workflows/sanity_checks_for_prs.yml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          DOCKER_HOST: docker_intentionally_disabled
+  commit_lint:
+    name: Ensure Git etiquette
+    runs-on: ubuntu-22.04
+    timeout-minutes: 5
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
```
Comment (tobiasleibner) [inline .github/workflows/sanity_checks_for_prs.yml:42]: Not a problem, just curious: Why do we use different python versions here and above?

File: .github/workflows/sanity_checks_for_prs.yml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        run: |
+          from github import Github
+          import os
+          import sys
+          repo_name = os.environ['GITHUB_REPOSITORY']
+          head_sha = os.environ['HEAD_SHA']
+          base_sha = os.environ['BASE_SHA']
+          g = Github(os.environ["GITHUB_TOKEN"])
+          blocked_words = ["fixup!", "squash!",]
+          blocked_messages = ["f", "REVERT ME BEFORE MERGE"]
```
Comment (tobiasleibner) [inline .github/workflows/sanity_checks_for_prs.yml:62]: Where does the "f" come from?

File: .github/workflows/sanity_checks_for_prs.yml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+          repo = g.get_repo(repo_name)
+          head_commit = repo.get_commit(head_sha)
+          base_commit = repo.get_commit(base_sha)
+          commits = repo.compare(base_commit.sha, head_commit.sha).commits
+          for commit in commits:
+              commit_msg = commit.commit.message
```
Comment (tobiasleibner) [inline .github/workflows/sanity_checks_for_prs.yml:68]: Is the `commit.commit.message` correct here? Looks like a commit too much on first sight.

File: .github/workflows/sanity_checks_for_prs.yml (hunk 1 above)
Comment (renefritze) [reply .github/workflows/sanity_checks_for_prs.yml:42]: Don't see any. Probably combined from two different actions in the before times?

File: .github/workflows/sanity_checks_for_prs.yml (hunk 2 above)
Comment (renefritze) [reply .github/workflows/sanity_checks_for_prs.yml:62]: No idea.

File: .github/workflows/sanity_checks_for_prs.yml (hunk 3 above)
Comment (renefritze) [reply .github/workflows/sanity_checks_for_prs.yml:68]: Sorry, I forgot to note that this a copy of https://github.com/arup-group/oasys-core-cmake/blob/main/.github/workflows/sanity_checks_for_prs.yml

Comment (tobiasleibner) [reply .github/workflows/sanity_checks_for_prs.yml:68]: Okay, so it is probably correct?

Comment (renefritze) [reply .github/workflows/sanity_checks_for_prs.yml:68]: Yeah. It's worked for good while for us I think :)

**Pull request merged**